doc/dart_documentation_comment_specification.md

```diff
@@ -0,0 +1,478 @@
+  void foo() {}
+}
+
+/// Usage in documentation:
+/// * [foo]   -> Resolves to the method foo().
+/// * [A.foo] -> Resolves to the constructor A.foo().
+
+```
+
+* **Getters and Setters:** A reference to a property name (e.g., `[value]`) resolves to the *conceptual property* rather than the individual getter or setter. See full discussion in [Section 6.2.2](#6.2.2.-getters-and-setters).
+
+* **Shadowing**: A name in an inner scope (like a `[parameterName]`) "shadows" a name in an outer scope (like a class field). The shadowed outer element must be qualified to be referenced, e.g., `[MyClass.fieldName]`.
+
+## **6\. Associating Documentation with Elements**
+
+While the placement rules in [Section 3](#3.-placement-of-documentation-comments) define where a doc comment must be written, this section defines how documentation tools should associate a specific documentation block with a given code element, especially in cases where the association isn't direct.
+
+### **6.1. The General Rule**
+
+The documentation for a given Dart element is, by default, the single doc comment that *immediately precedes* its declaration in the source code.
+
+### **6.2. Inheritance and Special Association Rules**
+
+In some cases, the effective documentation for an element is determined by context, inheritance, or combination rules rather than by a direct preceding comment.
+
+#### **6.2.1. Overridden Members**
+
+When a member overrides an ancestor, its documentation is determined by a set of inheritance and precedence rules.
+
+* **Explicit Documentation:** If an overriding member does have its own doc comment, that comment takes full precedence. The documentation from any base members is ignored.
+
+* **Unambiguous Documentation Inheritance** If a member overrides a member from an ancestor and does not have its own doc comment, it inherits the documentation from the base member. This rule applies when the source of the documentation is unambiguous (e.g., overriding a single concrete method).  Documentation tools should display this inherited comment, ideally noting its origin (e.g., "Copied from BaseClass").
+
+* **Ambiguous Documentation from Multiple Ancestors** The behavior for resolving documentation when a member inherits conflicting documentation from multiple ancestors is *reserved for future standardization*. Currently, tools may handle this case at their discretion, or ignore inherited documentation in ambiguous scenarios. It is recommended that tools *issue a warning* when this scenario is encountered, prompting the user to provide explicit documentation to resolve the ambiguity.
+
+#### **6.2.2. Getters and Setters**
+
+A getter and a setter that share the same name are treated as a single *conceptual property* rather than the individual getter or setter.
+
+Documentation tools should handle this property with the following rules:
+
+* **Display:** The getter and setter should be presented as a single item in the final documentation.
```

Comment:
Interesting, this is _not_ how DAS works today.

If you write `A().x = 1`, then the hover text in the IDE for `x =` is the doc comment for the setter, `set x`.

I think this is a good UX, and maybe this rule should instead be left to the discretion of the tool.